mesonbuild/compilers/cuda.py

```diff
@@ -384,19 +383,23 @@ def _to_host_flags(cls, flags: T.List[str], phase: _Phase = _Phase.COMPILER) ->
                 else:                            # -Isomething
                     val = flag[2:]
                 flag = flag[:2]                  # -I
-            elif flag in cls._FLAG_LONG2SHORT_WITHARGS or \
-                 flag in cls._FLAG_SHORT2LONG_WITHARGS:
+            elif flag in self._FLAG_LONG2SHORT_WITHARGS or \
+                 flag in self._FLAG_SHORT2LONG_WITHARGS:
                 # This is either -o or a multi-letter flag, and it is receiving its
                 # value isolated.
                 try:
                     val = next(flagit)           # -o something
                 except StopIteration:
                     pass
-            elif flag.split('=',1)[0] in cls._FLAG_LONG2SHORT_WITHARGS or \
-                 flag.split('=',1)[0] in cls._FLAG_SHORT2LONG_WITHARGS:
+            elif flag.split('=',1)[0] in self._FLAG_LONG2SHORT_WITHARGS or \
+                 flag.split('=',1)[0] in self._FLAG_SHORT2LONG_WITHARGS:
                 # This is either -o or a multi-letter flag, and it is receiving its
                 # value after an = sign.
                 flag, val = flag.split('=',1)    # -o=something
+            # Some dependencies (e.g., BoostDependency) add unspaced "-isystem/usr/include" arguments
+            elif flag.startswith('-isystem'):
```

Comment:
@nirbheek this finally fixes the issue from https://github.com/mesonbuild/meson/pull/7418#discussion_r452087151 properly